src/devices/Ina226/Ina226.cs

```diff
@@ -0,0 +1,579 @@
+    /// interface. The device monitors both a shunt voltage drop and bus supply voltage.
+    /// Programmable calibration value, conversion times, and averaging, combined with an
+    /// internal multiplier, enable direct readouts of current in ampers an power in watts.
+    /// <see href="https://www.ti.com/lit/ds/symlink/ina226.pdf"/>
+    /// </summary>
+    [Interface("INA226 Bidirectional Current/Power monitor")]
+    public class Ina226 : IDisposable
+    {
+        private I2cDevice _i2cDevice;
+        private bool _disposeI2cDevice = false;
+
+        // cache for commonly used configuration values
+        private ushort _calibrationValue;
+        private Ina226OperatingMode _operatingMode = Ina226OperatingMode.ShuntAndBusContinuous;
+        private Ina226SamplesAveraged _samplesAveraged = Ina226SamplesAveraged.Quantity_1;
+        private Ina226BusVoltageConvTime _busConvTime = Ina226BusVoltageConvTime.Time140us;
+        private Ina226ShuntConvTime _shuntConvTime = Ina226ShuntConvTime.Time140us;
+        private int _busConvTimeInt;
+        private int _shuntConvTimeInt;
+        private int _samplesAveragedInt;
+        private double _currentLsb;
+
+        /// <summary>
+        /// Construct an Ina226 device using an I2cDevice
+        /// </summary>
+        /// <remarks>
+        /// This binding does not dispose the passed in I2cDevice.
+        /// </remarks>
+        /// <param name="i2cDevice">The I2cDevice initialized to communicate with the INA226</param>
+        public Ina226(I2cDevice i2cDevice)
+        {
+            _i2cDevice = i2cDevice ?? throw new ArgumentNullException(nameof(i2cDevice));
+            _currentLsb = 1F;
+        }
+
+        /// <summary>
+        /// Construct an INA226 device useing I2cConnectionSettings passed in.
+        /// </summary>
+        /// <remarks>
+        /// This binding creates an I2cDevice for communication with the INA226. This I2cDevice is disposed when the INA226 is disposed.
+        /// </remarks>
+        /// <param name="settings">The I2cConnectionSettings object initialized with the appropriate settings to communicate with the INA226</param>
+        public Ina226(I2cConnectionSettings settings)
+            : this(I2cDevice.Create(settings)) => _disposeI2cDevice = true;
+
+        /// <summary>
+        /// Dispose of instance
+        /// </summary>
+        public void Dispose()
+        {
+            if (_disposeI2cDevice)
+            {
+                _i2cDevice?.Dispose();
+                _i2cDevice = null!;
+            }
+        }
+
+        /// <summary>
+        /// Converts the selected value for the bus voltage conversion time to an int
+        /// </summary>
+        /// <param name="convTimeEnum">Ina226BusVoltageConvTime enum</param>
+        /// <returns>Integer value corresponding to the conversion time enum param</returns>
+        internal static int BusConvTimeEnumToInt(Ina226BusVoltageConvTime convTimeEnum)
```

Comment:
Consider: BusConversionTimeEnumToMicroseconds

Reply:
note: for internal stuff it's only recommendation to fix this, I won't block this PR because of this. For public stuff it should be changed